external_search.py

```diff
@@ -48,99 +60,189 @@ def __init__(self, url=None, works_index=None):
             ExternalSearchIndex.__client = Elasticsearch(
                 url, use_ssl=use_ssl, timeout=20, maxsize=25
             )
-            ExternalSearchIndex.__client.works_index = works_index
+
+            # Set the index and the alias to the same value for
+            # now.
+            ExternalSearchIndex.__client.works_alias = works_index
             if not url:
                 raise Exception("Cannot connect to Elasticsearch cluster.")
 
-        self.works_index = self.__client.works_index
         self.indices = self.__client.indices
         self.search = self.__client.search
         self.index = self.__client.index
         self.delete = self.__client.delete
         self.exists = self.__client.exists
+
+        # Search queries run against the works_alias.
+        self.works_alias = self.__client.works_alias
+
+        # Document upload runs against the works_index.
+        self.set_works_index(self.works_alias)
+
         def bulk(docs, **kwargs):
             return elasticsearch_bulk(self.__client, docs, **kwargs)
         self.bulk = bulk
-            
+
+    def set_works_index(self, current_alias):
+        """Finds or creates the index based on provided configuration"""
+        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
+        found = not (index_details.get('status')==404 or 'error' in index_details)
+
+        def _set_works_index(name):
+            self.works_index = self.__client.works_index = name
+
+        if found:
+            # We found an index for the alias in configuration. Assume
+            # there is only one.
+            _set_works_index(index_details.keys()[0])
+
+        if current_alias.endswith(self.CURRENT_ALIAS_SUFFIX):
+            # The alias culled from configuration is intended to be
+            # a current alias. Get the index name from here.
+            base_works_index = self._base_works_index(current_alias)
+            version = ExternalSearchIndexVersions.latest()
+            _set_works_index(base_works_index+'-'+version)
+        else:
+            # Without the CURRENT_ALIAS_SUFFIX, assume the index string
+            # from config is the index itself and needs to be swapped.
+            _set_works_index(current_alias)
```

Comment:
That sounds good for 2. I'd like to have a migration script to create the alias for any apps that are still running with an index in their config. They'll have to make a config change, but I don't think there are any in production, unless brooklyn. Hmm...